### DISCUSSION TITLE
Arm64 - RaspberryPI - Kubernetes

### DISCUSSION THREAD

**Discussion by dhenry123** [healthchecks/healthchecks#655] [General]
Hi i ve just complete the installation on kubernetes cluster  raspberry PI 4  , i m not a python developper, so advices are welcome to change something wrong.

image was built from alpine

Dockerfile (only postgresql support): 
```
FROM alpine:latest
RUN apk add --no-cache gcc python3-dev libpq-dev git libc-dev libffi-dev bash py-pip
RUN ln -s /usr/bin/python3 /usr/bin/python
RUN mkdir -p /webapps
WORKDIR /webapps
RUN git clone https://github.com/healthchecks/healthchecks.git
RUN python3 -m venv hc-venv \
    && source hc-venv/bin/activate \
    && python3 -m pip install --upgrade pip \
    && pip3 install wheel \
    && pip3 install -r healthchecks/requirements.txt \
    && pip3 install uwsgi tzdata \
    && rm -f healthchecks/hc/local_settings.py \
    && DEBUG=False SECRET_KEY=build-key healthchecks/manage.py collectstatic --noinput \
    && DEBUG=False SECRET_KEY=build-key healthchecks/manage.py compress
RUN addgroup -S hc && adduser -S hc -G hc
ENV PYTHONUNBUFFERED=1
## To fix the django-csrf bug
## source : https://stackoverflow.com/questions/70508568/django-csrf-trusted-origins-not-working-as-expected
COPY ./updatesetting /
RUN cat /updatesetting >> healthchecks/hc/settings.py
## entrypoint
COPY ./docker-entrypoint.sh /
RUN chmod +x /docker-entrypoint.sh
USER hc
EXPOSE 8000
ENTRYPOINT ["/docker-entrypoint.sh"]
```
You could see i fix a crsf django problem for the POC and 
It's possible to optimize the docker image with "AS" and "FROM", but not enough time

updatesetting file content :
```
CSRF_TRUSTED_ORIGINS = os.getenv("CSRF_TRUSTED_ORIGINS", "").split(",")
```
Add CSRF_TRUSTED_ORIGIN="https://your site" as parameter of container...
and TIMEZONE is hard coded in settings (UTC)

docker-entrypoint.sh
```
#!/bin/bash
source hc-venv/bin/activate
hc-venv/bin/uwsgi /uwgsi-config/uwsgi.ini
```
i ve set uwgsi with 2 workers, may be one will be ok (i ll check), limitRange : cpu max: 2000m et mem max: 300Mi, with no jobs state is avg : cpu 80m, ram 159Mi and the same not enough time at this moment but with read only security image context, i ve no network...

You could see the entire POC at https://www.mytinydc.com/en/kubernetes-healthchecks/

**Something to set for horizontal scaling** ???
 
Thanks for your work
+++

**Comment by cuu508** [comment]
1. You should not need to set CSRF_TRUSTED_ORIGINS if the reverse proxy is sending a "X-Forwarded-Proto" header. See the section "TLS Termination" here: https://healthchecks.io/docs/self_hosted_docker/
And also the discussion here: https://github.com/healthchecks/healthchecks/pull/597

2. Instead of patching `settings.py` I would suggest to put any needed customizations in `local_settings.py` (`settings.py` looks for the existence of `local_settings.py` and imports it if found, [source](https://github.com/healthchecks/healthchecks/blob/09a99d3e9c8329d20505e058a6514ba94cb17bb7/hc/settings.py#L287))

3. Consider using the official [Dockerfile](https://github.com/healthchecks/healthchecks/tree/master/docker) and [images](https://hub.docker.com/r/healthchecks/healthchecks).

**Reply by dhenry123** [reply]
Thank you for your answer.
Best regards.
